web/crux/prisma/schema.prisma

```diff
@@ -565,6 +566,57 @@ model Storage {
   @@unique([name, teamId])
 }
 
+enum PipelineTypeEnum {
+  github
+  gitlab
+  azure
+}
+
+enum PipelineStatusEnum {
+  unknown
+  queued
+  running
+  successful
+  failed
+}
+
+model Pipeline {
+  id          String           @id @default(uuid()) @db.Uuid
+  createdAt   DateTime         @default(now()) @db.Timestamptz(6)
+  createdBy   String           @db.Uuid
+  updatedAt   DateTime         @updatedAt @db.Timestamptz(6)
+  updatedBy   String?          @db.Uuid
+  name        String           @db.VarChar(70)
+  type        PipelineTypeEnum
+  description String?
+  icon        String?
```

Comment:
Always has an icon, doesn't it?

Reply:
No. If the user doesn't choose one, we will display the logo of the pipeline type.